x-pack/legacy/plugins/maps/public/angular/map_controller.js

```diff
@@ -124,6 +126,77 @@ app.controller('GisMapController', ($scope, $route, kbnUrl, localStorage, AppSta
     globalState: globalState,
   });
 
+  /* Saved Queries */
+  $scope.showSaveQuery = capabilities.get().maps.saveQuery;
+
+  $scope.$watch(() => capabilities.get().maps.saveQuery, (newCapability) => {
+    $scope.showSaveQuery = newCapability;
+  });
+
+  $scope.onQuerySaved = savedQuery => {
+    $scope.savedQuery = savedQuery;
+  };
+
+  $scope.onSavedQueryUpdated = savedQuery => {
+    $scope.savedQuery = { ...savedQuery };
```

Comment:
How do you trigger this callback?

Reply:
@nreese [onSavedQueryUpdated](https://github.com/elastic/kibana/pull/44442/files#diff-b41718c37611de3cc6acffc1c56a3fe9R24) is handed down through the maps directive to the saved query management component. It handles changes that are made to the saved query once it's loaded into the search bar.

Reply:
ok, but what actions does a user take to trigger the callback? I tried changing the search but that did not call this callback

Reply:
If you change the saved query and not save it, then click on it again in the saved query management component, the callback should trigger. We specifically make a copy of the saved query so that we can effectively [reset](https://github.com/elastic/kibana/pull/43927) it if it's not saved before reloading it.